table/partitioned_filter_block_test.cc

```diff
@@ -66,12 +66,13 @@ class PartitionedFilterBlockTest : public testing::Test {
   }
 
   uint64_t MaxFilterSize() {
+    uint32_t dont_care1, dont_care2;
```

Comment:
Style-wise it looked nicer to me. Let me know if you want me to put it on the next line.

This is in test so the cons of dynamic_cast are less problematic and we can yet benefit from its pros. Lets say that we switch to static_cast and a later code change invalidates our assumption about the type. How would the future rocksdb developer would be notified of that?